Allow initializing Distance from angle-like parallax (#15712)
astropy/astropy: docs/changes/coordinates/15712.bugfix.rst

> Fix a bug in the coordinates ``Distance`` class to allow initialization with
> ``parallax`` as any angle-like value. This includes types like ``Column`` which have a
> unit but are not ``Quantity`` objects.

Reviewer: Saying that a bug in `coordinates` is fixed is redundant because the change log entry is in the `coordinates` bug fixes section.
```suggestion
``Distance`` now accepts as ``parallax`` any angle-like value.
This includes types like ``Column`` which have a unit but are not ``Quantity`` subclasses.
```
